qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,261 @@
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return gaussian(duration=self.duration, amp=self.amp,
+                        sigma=self.sigma, zero_ends=False)
+
+
+class GaussianSquare(ParametricPulse):
+    """
+    A square pulse with a Gaussian shaped risefall on either side:
+
+        risefall = duration - width / 2
+
+    0 <= x < risefall
+        f(x) = amp * exp( -(1/2) * (x - risefall/2)^2 / sigma^2) )
+
+    risefall <= x < risefall + width
+        f(x) = amp
+
+    risefall + width <= x < duration
+        f(x) = amp * exp( -(1/2) * (x - (risefall + width)/2)^2 / sigma^2) )
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 width: int):
+        """Initialize the gaussian square command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Gaussian and of the square pulse.
+            sigma: A measure of how wide or narrow the Gaussian risefall is; described
+                   mathematically in the class docstring.
+            width: The duration of the embedded square pulse.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._width = width
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    @property
+    def width(self):
+        return self._width
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return gaussian_square(duration=self.duration, amp=self.amp,
+                               risefall=(self.duration - self.width) / 2, sigma=self.sigma,
+                               zero_ends=False)
+
+
+class Drag(ParametricPulse):
+    """
+    A pulse whose envelope is shaped by a drag pulse. This is so named by the technique
+    Derivative Removal by Adiabatic Gate (DRAG). It is constructed to reduce the frequency
+    spectrum of a normal gaussian curve near the |1>-|2> transition.
+
+        f(x) = Gaussian + 1j * beta * d/dx [Gaussian]
+             = Gaussian + 1j * beta * (-x / sigma) [Gaussian]
+
+    where 'Gaussian' is:
+        Gaussian(x, amp, sigma) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )
+
+    Ref:
+        [1] Gambetta, J. M., Motzoi, F., Merkel, S. T. & Wilhelm, F. K.
+        Analytic control methods for high-fidelity unitary operations
+        in a weakly nonlinear oscillator. Phys. Rev. A 83, 012308 (2011).
+    """
```

Comment:
Nitpicking more than anything else but perhaps the PRL is also good: https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.103.110501

Reply:
Add both, you think?